SYMBOL_TABLE_ENTRY_CREATION.md

```diff
@@ -0,0 +1,354 @@
+    
+        var midpts: shared SymEntry(int);
+        var values: shared SymEntry(string);
+        
+        proc init(entryName:string, entry:borrowed NewSymEntry) {
+            name = entryName;
+            composite = entry;
+            
+            midpts = composite.midptEntry;
+            values = composite.valuesEntry;
+        }
+    }
+}
+```
+
+Now that we have our object, we need to add methods to create it from Chapel arrays. These methods will create the required `SymEntry` objects and use them to generate a `NewSymEntry` object. Let's update our module to add this process.
+
+```chapel
+module NewSymObject {
+    proc getNewSymObj(midpts: [] int, values: [] string, st: borrowed SymTab): owned NewSymObj throws {
+        var midptEntry = new shared SymEntry(midpts);
+        var valuesEntry = new shared SymEntry(values);
+        var newEntry = new shared NewSymEntry(midptEntry, valuesEntry);
+        var name = st.nextName();
+        st.addEntry(name, segEntry);
+        return getNewSymObj(name, st, segEntry.etype);
+    }
+    
+    class NewSymObj {
+        var name: string;
+    
+        var composite: borrowed NewSymEntry;
+    
+        var midpts: shared SymEntry(int);
+        var values: shared SymEntry(string);
+        
+        proc init(entryName:string, entry:borrowed NewSymEntry) {
+            name = entryName;
+            composite = entry;
+            
+            midpts = composite.midptEntry;
+            values = composite.valuesEntry;
+        }
+    }
+}
+```
+
+Notice that we have a call to `getNewSymObj()`. This overload call allows us to build the `NewSymObj` from the name of the `NewSymEntry`. This is important because it allows us to access the borrowed object from the name. This will be important when we add messages to perform various operations on an already created object because it allows easy access of the object. Let's now build this overload function in our module.
+
+```chapel
+module NewSymObject {
+    proc getNewSymObj(name: string, st: borrowed SymTab): owned NewSymObj throws {
+        var abstractEntry = st.lookup(name);
+        if !abstractEntry.isAssignableTo(SymbolEntryType.NewSymEntry) {
+            var errorMsg = "Error: Unhandled SymbolEntryType %s".format(abstractEntry.entryType);
+            saLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            throw new Error(errorMsg);
+        }
+        var entry: NewSymEntry = abstractEntry: borrowed NewSymEntry();
+        return new owned NewSymObj(name, entry);
+    }
+    proc getNewSymObj(midpts: [] int, values: [] string, st: borrowed SymTab): owned NewSymObj throws {
+        var midptEntry = new shared SymEntry(midpts);
+        var valuesEntry = new shared SymEntry(values);
+        var newEntry = new shared NewSymEntry(midptEntry, valuesEntry);
+        var name = st.nextName();
+        st.addEntry(name, segEntry);
+        return getNewSymObj(name, st, segEntry.etype);
```

Comment:
same `newEntry` instead of `segEntry` comment

Reply:
updated